src/resolver/select_resolver.h

```diff
@@ -76,7 +76,6 @@ class SelectResolver {
     project_references_ =
         std::make_shared<std::vector<std::shared_ptr<ProjectReference>>>();
 
-    join_predicates_->reserve(5);
     agg_references_->reserve(5);
     group_by_references_->reserve(5);
```

Comment:
why 5? add a comment to explain the magic number. Or better make it into a defined constant that can easily changed